src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const act of actions) {
+    let modifiedState = { ...currentState };
+
+    switch (act.type) {
+      case 'addProperties':
+        Object.assign(modifiedState, act.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of act.keysToRemove) {
+          delete modifiedState[key];
+        }
+        break;
+
+      case 'clear':
+        modifiedState = {};
+        break;
+
+      default:
+        return 'error';
```

Comment:
Returning a string 'error' in the default case is problematic because it changes the expected return type of the function from an array of objects to a string. Consider handling unknown action types by logging an error or simply continuing the loop without modifying the state.